File: Problems/No.77_Combinations/combinations.py
class Solution(object):
    def combine(self, n, k):
        """
        :type n: int
        :type k: int
        :rtype: List[List[int]]
        """

        # create combinations
        def create_combinations(start):
            if (len(combination) == k):
                comb = combination[:]
                combinations.append(comb)
            else:
                for i in range(start, n+1):
                    combination.append(i)
                    create_combinations(i+1)
                    combination.pop()

        # parameters
        combinations, combination = [], []

        # start computing
        create_combinations(1)
        
        return combinations

File: Problems/No.77_Combinations/combinations.cpp
class Solution {
public:

    // create combinations
    void create_combinations(int start, int end, int size, std::vector<int> &comb, std::vector<std::vector<int>> &combs) {

        if (comb.size() == size) combs.push_back(comb);
        else {

            for (int i=start; i<=end; i++) {

                comb.push_back(i);
                create_combinations(i+1, end, size, comb, combs);
                comb.pop_back();

            }

        }

    }

    vector<vector<int>> combine(int n, int k) {
        
        // parameters
        std::vector<std::vector<int>> combinations;
        std::vector<int> combination;

        // start computing
        create_combinations(1, n, k, combination, combinations);

        return combinations;

    }
};
